reader.go

```diff
@@ -792,6 +799,11 @@ func (this *PdfReader) findXref() error {
 			this.xrefPos = result
 			break
 		}
+
+		// (token, err) == ("", nil) indicates EOF reached before finding "startxref"
+		if token == "" {
```

Comment:
The code changes look good. Is it possible to write a unit test for this case when the token is empty? I don't know how hard it is to mock pdf reading in a test. If it is not straightforward, we can move on.

Reply:
That's a good call out.